server/sonar-web/src/main/webapp/WEB-INF/app/controllers/project_controller.rb

```diff
@@ -232,7 +232,7 @@ def delete_snapshot_history
 
     sid = params[:snapshot_id]
     if sid
-      Snapshot.update_all("status='U'", ["id=? or root_snapshot_id=(?)", sid.to_i, sid.to_i])
+      Snapshot.update_all("status='U'", ["id=?", sid.to_i])
```

Comment:
do we still need "update_all", isn't there a "update"?
shouldn't you use the uuid rather than the id? (in preparation for the day we will remove ids)